sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -7,184 +7,168 @@
+    if referenced_client:
+        return referenced_client.get_secret(key_vault_identifier.name, version=key_vault_identifier.version).value
 
-        if config.secret_id is None:
-            raise AttributeError("Key Vault reference must have a uri value.")
+    if key_vault_options.secret_resolver is not None:
+        return key_vault_options.secret_resolver(config.secret_id)
 
-        key_vault_identifier = KeyVaultSecretIdentifier(config.secret_id)
+    raise AttributeError(
+        "No Secret Client found for Key Vault reference %s" % (key_vault_identifier.vault_url)
+    )
 
-        referenced_client = next(
-            (client for client in secret_clients if client.vault_url == key_vault_identifier.vault_url), None
-        )
+def __is_json_content_type(content_type):
+    if not content_type:
+        return False
 
-        if referenced_client is None and key_vault_options.credential is not None:
-            referenced_client = SecretClient(
-                vault_url=key_vault_identifier.vault_url, credential=key_vault_options.credential
-            )
-            secret_clients[key_vault_identifier.vault_url] = referenced_client
-
-        if referenced_client:
-            try:
-                return referenced_client.get_secret(
-                    key_vault_identifier.name, version=key_vault_identifier.version
-                ).value
-            except ResourceNotFoundError:
-                raise KeyVaultReferenceError(
-                    "Key Vault %s does not contain secret %s"
-                    % (key_vault_identifier.vault_url, key_vault_identifier.name)
-                )
-
-        if key_vault_options.secret_resolver is not None:
-            return key_vault_options.secret_resolver(config.secret_id)
-        raise KeyVaultReferenceError(
-            "No Secret Client found for Key Vault reference %s" % (key_vault_identifier.vault_url)
-        )
+    content_type = content_type.strip().lower()
+    mime_type = content_type.split(";")[0].strip()
 
-    @staticmethod
-    def __is_json_content_type(content_type):
-        if not content_type:
-            return False
+    type_parts = mime_type.split("/")
+    if len(type_parts) != 2:
+        return False
 
-        content_type = content_type.strip().lower()
-        mime_type = content_type.split(";")[0].strip()
+    (main_type, sub_type) = type_parts
+    if main_type != "application":
+        return False
 
-        type_parts = mime_type.split("/")
-        if len(type_parts) != 2:
-            return False
+    sub_types = sub_type.split("+")
+    if "json" in sub_types:
+        return True
 
-        (main_type, sub_type) = type_parts
-        if main_type != "application":
-            return False
+    return False
 
-        sub_types = sub_type.split("+")
-        if "json" in sub_types:
-            return True
+class AzureAppConfigurationProvider:
```

Comment:
Given that this object holds an instance of a client, we will need to expose this as a context manger and allow it to be closeable (such that the connection can be cleaned up).
This means that we will probably need to still keep the object public so that it can be properly documented....

Reply:
Well, this would get a bit awkward when it comes to refresh. Typically it's best to reuse the same clients as it is a costly operation to create new ones.

Also, how would this work given we have one to many clients, for App Configuration (will be with geo-replication) and Key Vault.